front/app/modules/commercial/insights/services/insightsDetectCategories.ts

```diff
@@ -0,0 +1,22 @@
+import { API_PATH } from 'containers/App/constants';
+import streams, { IStreamParams } from 'utils/streams';
+
+const getInsightsDetectCategoriesEndpoint = (viewId: string) =>
+  `insights/views/${viewId}/detected_categories`;
+
+export type IInsightsDetectedCategoriesData = { names: string[] };
+export interface IInsightsDetectedCategories {
+  data: IInsightsDetectedCategoriesData;
+}
+
+export function insightsDetectedCategoriesStream(
+  insightsViewId: string,
+  streamParams: IStreamParams | null = null
```

Comment:
I think you should disable caching for this one, since BE filters out the existing categories from the result